packages/server/src/db/index.js

```diff
@@ -284,6 +285,20 @@ function deleteKeyword(id) {
         .del();
 }
 
+async function getNewGrantsForAgency(agency) {
+    const agencyCriteria = await getAgencyCriteriaForAgency(agency.id);
+
+    const rows = await knex(TABLES.grants)
+        .select(knex.raw(`${TABLES.grants}.*, count(*) OVER() AS total_grants`))
```

Comment:
@TylerHendrickson thanks for pushing on this! I was able to ensure that we only returned at most 3 grants while also getting a count of the rows directly from postgres in a single query.

I implemented it as a window-function and will take a next step once we are using the feature to analyze the query a bit more to see how postgres is processing this.